pkg/planner/cardinality/pseudo.go

```diff
@@ -243,3 +245,52 @@ func getPseudoRowCountByColumnRanges(tc types.Context, tableRowCount float64, co
 	}
 	return rowCount, nil
 }
+
+// getPseudoRowCountWithPartialStats calculates the row count if there are no statistics on the index, but there are column stats available.
+func getPseudoRowCountWithPartialStats(sctx planctx.PlanContext, coll *statistics.HistColl, indexRanges []*ranger.Range,
```

Comment:
nit: feels like this method can be a in better place than pseudo.go. This file seems to contain methods with only complete guess, however getPseudoRowCountWithPartialStats is potentially using GetRowCountByColumnRanges, maybe it can be in row_count_column.go? Just a thought.